File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
 
public class Solution {
    public ListNode ReverseList(ListNode head) {
        ListNode prev = null, cur = head != null ? new ListNode(head.val, head.next) : null;
        while (cur != null)
        {
            ListNode tmp = cur.next != null ? new ListNode(cur.next.val, cur.next.next) : null;
            cur.next = prev;
            prev = cur;
            cur = tmp;
        }
        return prev;
    }
}

File: Data Structures & Algorithms/linked-list-cycle-detection/submission-0.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func hasCycle(head *ListNode) bool {
	if head == nil || head.Next == nil {
		return false
	}

    fast, slow := head, head
	for {		
		slow = slow.Next
		fast = fast.Next
		if fast == nil {
			return false
		}

		fast = fast.Next
		if fast == nil {
			return false
		} else if fast == slow {
			return true
		}
	}
}
